### PR TITLE
Remove deprecated CategoryManager

### PR DESCRIPTION
**Pull request by ClementGautier**
# Goal
The goal here is to remove the deprecated CategoryManager before tagging the 1.5

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N/A
| Added Behats                      | N/A
| Changelog updated                 | Y
| Review and 2 GTM                  | todo
| Micro Demo to the PO (Story only) | N/A
| Migration script                  | N/A
| Tech Doc                          | N/A

### CODE REVIEW

File: CHANGELOG-1.5.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# 1.5.x
+
+## BC breaks
+
+- Change constructor of `Pim\Bundle\EnrichBundle\Controller\ProducctController` to remove `Pim\Bundle\CatalogBundle\Manager\CategoryManager`
+- Change interface `Akeneo\Component\Classification\Repository\CategoryRepositoryInterface` to add the methods `getFilledTree` and `getTreeChoices` previously on `Pim\Bundle\CatalogBundle\Manager\CategoryManager`
+- Removed `Pim\Bundle\CatalogBundle\Manager\CategoryManager`
```
Comment (phaseinducer) [inline CHANGELOG-1.5.md]: *remove

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php [hunk 2]
```diff
@@ -397,4 +413,29 @@ public function getOrderedAndSortedByTreeCategories()
 
         return $queryBuilder->getQuery()->getResult();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilledTree(CategoryInterface $root, Collection $categories)
```
Comment (phaseinducer) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: Not sure removing a manager just by moving its methods inside the repo is the good solution. It breaks SRP, your repo becomes the new manager, nothing is solved imho.

Comment (ClementGautier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: Actually it now respect SRP. Manager did multiple things (fetching, saving, etc). This method is a "fetching" method so its the responsability of the repository to do it.

Comment (ClementGautier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: Well as we do not have a "Finder" layer, Repositories are used to do that. Actually if we had "Finder", "Saver", "Updater" (that actually merge datas, so I would more likely call them Mergers) and "Remover" we could get rid of the repositories but it's not the case at the moment.

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php [hunk 3]
```diff
@@ -346,6 +347,21 @@ public function getTrees()
     /**
      * {@inheritdoc}
      */
+    public function getTreeChoices()
```
Comment (solivier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: if it's only used by datagrid you could move this method into a dedicated repository

File: src/Akeneo/Component/Classification/Repository/CategoryRepositoryInterface.php [hunk 4]
```diff
@@ -164,6 +164,11 @@ public function getPath($node);
     public function getTrees();
 
     /**
+     * @return array
```
Comment (solivier) [inline src/Akeneo/Component/Classification/Repository/CategoryRepositoryInterface.php]: one line comment

**Comment by solivier** [comment]
gtm after comment fixed

File: src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php (hunk 3 above)
Comment (ClementGautier) [inline src/Akeneo/Bundle/ClassificationBundle/Doctrine/ORM/Repository/CategoryRepository.php]: I'll not do a specific repository per usage. The repository meaning is to group all functions that access the storage layer for a document/entity. I don't see the benefit ofhaving one for datagrid usage as, basicly, we could use this function in other contexts.

File: CHANGELOG-1.5.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+# 1.5.x
+
+## BC breaks
+
+- Change constructor of `Pim\Bundle\EnrichBundle\Controller\ProducctController` to remove `Pim\Bundle\CatalogBundle\Manager\CategoryManager`
```
Comment (skeleton) [inline CHANGELOG-1.5.md]: Produ**c**ctController

**Comment by jjanvier** [comment]
gtm

**Comment by skeleton** [comment]
gtm once fixed

**Comment by ClementGautier** [comment]
1.5 is about to be released and no more "nice to have" BC breaks are allowed. Please reopen this PR on the master branch for the upcoming 1.6.

**Pull request closed without merging**